drivers/clock_control/clock_control_it8xxx2.c

```diff
@@ -0,0 +1,334 @@
+
+LOG_MODULE_REGISTER(clock_control_it8xxx2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#define PLLFREQ_MASK 0x0F
+
+struct clock_control_it8xxx2_config {
+	struct ecpm_it8xxx2_regs *reg_base;
+	int pll_freq;
+	uint32_t cpu_clock;
+};
+
+const struct clock_control_it8xxx2_config *clock_config;
+
+/* ITE IT8XXX2 ECPM Functions */
+static inline int it8xxx2_clock_control_on(const struct device *dev,
+					clock_control_subsys_t sub_system)
+{
+	const struct clock_control_it8xxx2_config *cfg = dev->config;
+	struct ecpm_it8xxx2_regs *const ecpm_regs = cfg->reg_base;
+	struct it8xxx2_clock_control_cells *clk_cfg =
+		(struct it8xxx2_clock_control_cells *)(sub_system);
+
+	uint8_t offset = clk_cfg->offset;
+	uint8_t code = clk_cfg->bitmask;
+	uint8_t always = clk_cfg->always;
+
+	volatile uint8_t *reg =
+		(volatile uint8_t *)((uint32_t)ecpm_regs + (uint32_t)offset);
+
+	*reg = (*reg & ~code) | always;
+
+	return 0;
+}
+
+static inline int it8xxx2_clock_control_off(const struct device *dev,
+					clock_control_subsys_t sub_system)
+{
+	const struct clock_control_it8xxx2_config *cfg = dev->config;
+	struct ecpm_it8xxx2_regs *const ecpm_regs = cfg->reg_base;
+	struct it8xxx2_clock_control_cells *clk_cfg =
+		(struct it8xxx2_clock_control_cells *)(sub_system);
+
+	uint8_t offset = clk_cfg->offset;
+	uint8_t code = clk_cfg->bitmask;
+	uint8_t always = clk_cfg->always;
+
+	volatile uint8_t *reg =
+		(volatile uint8_t *)((uint32_t)ecpm_regs + (uint32_t)offset);
+
+	*reg |= (always | code);
+
+	return 0;
+}
+
+const uint32_t pll_reg_to_freq[8] = {MHZ(8), MHZ(16), MHZ(24), MHZ(32),
+				MHZ(48), MHZ(64), MHZ(72), MHZ(96)};
+/*
+ *  TODO: So far it only supports get PLL frequency but we'll add more
+ *  returned options in the future for available device nodes. In the future,
+ *  the related properties and nodes will be added in the it8xxx2.dtsi and
+ *  YAML file.
+ */
+static int it8xxx2_clock_control_get_rate(const struct device *dev,
+					  clock_control_subsys_t sub_system,
+					  uint32_t *rate)
+{
+	const struct clock_control_it8xxx2_config *const config = dev->config;
+	struct ecpm_it8xxx2_regs *const ecpm_regs = config->reg_base;
+
+	int reg_val = ecpm_regs->PLLFREQR & PLLFREQ_MASK;
+
+	*rate = pll_reg_to_freq[reg_val];
+
+	return 0;
+}
+
+/*
+ *  TODO: So far it only supports set CPU frequency but we'll add more
+ *  returned options in the future for available device nodes. In the future,
+ *  the related properties and nodes will be added in the it8xxx2.dtsi and
+ *  YAML file.
+ */
+static int it8xxx2_clock_control_set_rate(const struct device *dev,
+					  clock_control_subsys_t sub_system,
+					  clock_control_subsys_rate_t rate)
+{
+
+	const struct clock_control_it8xxx2_config *const config = dev->config;
+	struct ecpm_it8xxx2_regs *const ecpm_regs = config->reg_base;
+
+	int reg_val = ecpm_regs->PLLFREQR & PLLFREQ_MASK;
+	uint8_t mcu_clock_freq_select_curr = ecpm_regs->SCDCR0 & 0x07;
```

Comment:
Move `0x07` to a define (and possibly use `BIT_MASK`, if appropriate).

Reply:
@fkokosinski 

Thanks for guidance, and now it's updated as follows:
`uint8_t mcu_clock_freq_select_curr = ecpm_regs->SCDCR0 & MCUFREQ_MASK;`